Xamarin.Forms.Core/Markup/ElementGesturesExtensions.cs

```diff
@@ -0,0 +1,100 @@
+
+		/// <summary>Ensure that <typeparamref name="TGestureElement"/> has a <see cref="ClickGestureRecognizer"/>,
+		/// and pass it to the supplied <paramref name="init"/> Action</summary>
+		public static TGestureElement ClickGesture<TGestureElement>(this TGestureElement gestureElement, Action<ClickGestureRecognizer> init)
+			where TGestureElement : Element, IGestureRecognizers
+			=> Gesture(gestureElement, init);
+
+		/// <summary>Ensure that <typeparamref name="TGestureElement"/> has a <see cref="PanGestureRecognizer"/>,
+		/// and pass it to the supplied <paramref name="init"/> Action</summary>
+		public static TGestureElement PanGesture<TGestureElement>(this TGestureElement gestureElement, Action<PanGestureRecognizer> init)
+			where TGestureElement : Element, IGestureRecognizers
+			=> Gesture(gestureElement, init);
+
+		/// <summary>Ensure that <typeparamref name="TGestureElement"/> has a <see cref="PinchGestureRecognizer"/>,
+		/// and pass it to the supplied <paramref name="init"/> Action</summary>
+		public static TGestureElement PinchGesture<TGestureElement>(this TGestureElement gestureElement, Action<PinchGestureRecognizer> init)
+			where TGestureElement : Element, IGestureRecognizers
+			=> Gesture(gestureElement, init);
+
+		/// <summary>Ensure that <typeparamref name="TGestureElement"/> has a <see cref="SwipeGestureRecognizer"/>,
+		/// and pass it to the supplied <paramref name="init"/> Action</summary>
+		public static TGestureElement SwipeGesture<TGestureElement>(this TGestureElement gestureElement, Action<SwipeGestureRecognizer> init)
+			where TGestureElement : Element, IGestureRecognizers
+			=> Gesture(gestureElement, init);
+
+		/// <summary>Ensure that <typeparamref name="TGestureElement"/> has a <see cref="TapGestureRecognizer"/>,
+		/// and pass it to the supplied <paramref name="init"/> Action</summary>
+		public static TGestureElement TapGesture<TGestureElement>(this TGestureElement gestureElement, Action<TapGestureRecognizer> init)
+			where TGestureElement : Element, IGestureRecognizers
+			=> Gesture(gestureElement, init);
+
+		/// <summary>Ensure that <typeparamref name="TGestureElement"/> has a <typeparamref name="TGestureRecognizer"/>,
+		/// and pass it to the supplied <paramref name="init"/> Action</summary>
+		public static TGestureElement Gesture<TGestureElement, TGestureRecognizer>(
+			this TGestureElement gestureElement,
+			Action<TGestureRecognizer> init
+		) where TGestureElement : Element, IGestureRecognizers
+		  where TGestureRecognizer : GestureRecognizer, new()
+		{
+			init.Invoke(Ensure<TGestureRecognizer>(gestureElement.GestureRecognizers));
+			return gestureElement;
+		}
+
+		static TGestureRecognizer Ensure<TGestureRecognizer>(IList<IGestureRecognizer> gestureRecognizers)
+			where TGestureRecognizer : GestureRecognizer, new()
+		{
+			var gestureRecognizer = (TGestureRecognizer)gestureRecognizers.FirstOrDefault(g => g is TGestureRecognizer);
```

Comment:
I estimated that adding multiple gesture recognizers of the same type on a view would be rare.

The reason for reconfiguring an existing recognizer is when you want to reuse a view instance in another place and then reconfigure some gesture recognizer properties.
However, this seems to be an even more rare situation.

Unless devs watching this conversation indicate otherwise, I am inclined to simplify the implementation to always add a recognizer.

Reply:
I changed the implementation of the gesture helpers to always add a new instance. I also updated doc in PR description.